synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -125,28 +110,42 @@ def __init__(self, hs: "HomeServer"):
         # Whether to support MSC3772 is supported.
         self._relations_match_enabled = self.hs.config.experimental.msc3772_enabled
 
+    @lru_cache()
+    async def get_rules(self, event: EventBase) -> Dict[str, List[Dict[str, dict]]]:
```

Comment:
The caching here isn't quite right. `lru_cache` doesn't have async support, and we'll want to cache this with a key of `room_id` rather than an `event` object. 

```suggestion
    @cached()
    async def get_rules(self, room_id: str) -> Dict[str, List[Dict[str, dict]]]:
```

Reply:
I think it is what you need, I think the following would do it:

```suggestion
    @cached(uncached_args=("event",))
    async def get_rules(self, room_id: str, event: EventBase) -> Dict[str, List[Dict[str, dict]]]:
```

It is then clearer to replace `event.room_id` internally with just `room_id`, but those should be equivalent?

How does this break the code?

Reply:
NVMD it seems to work when I add a cache context